torchdata/dataloader2/communication/iter.py

```diff
@@ -145,6 +145,15 @@ def DataPipeBehindQueues(source_datapipe, protocol, blocking_request_get=False):
             protocol.response_reset_epoch()
 
         elif isinstance(request, communication.messages.ResetIteratorRequest):
+            # Ensure only reset iterator once for the dispatching process
+            if reset_iterator_counter is not None:
+                reset_iterator_counter.increment()
+                while not reset_iterator_counter.is_reached():
+                    yield True
+                # Sync between loops within the dispatching process
+                source_datapipe.reset_iterator()
+                yield True
+                reset_iterator_counter.reset()
```

Comment:
Change 3

Reply:
Just to confirm I understand - this is to handle the situation where some workers are handling `GetNextRequest` while some are trying to `reset`? You want all `GetNext` to be done before the dispatching process executes `reset`?

Reply:
Not workers. It only happens to the dispatching process when multiple leaf DataPipes shares the same data source (Round robin demux on the same DataPipe) in a single process.
It handles the case when some loops have received `reset` while the others haven't. We want to wait to request `getNext` until all loops have received `reset`. Otherwise, there will be a case that the data source is reset during the middle of iteration for other loops